addons/web_editor/static/src/components/media_dialog/image_selector.js

```diff
@@ -162,7 +162,24 @@ export class ImageSelector extends FileSelector {
 
     validateUrl(...args) {
         const { isValidUrl, path } = super.validateUrl(...args);
-        const isValidFileFormat = IMAGE_EXTENSIONS.some(format => path.endsWith(format));
+        // Validate that the URL is an image
+        const isValidImage = async () => {
+            fetch(path, { method: 'HEAD' })
+            .then(response => {
+                const contentType = response.headers.get('content-type');
+                return contentType && IMAGE_MIMETYPES.includes(contentType);
+            }, () => {
+                return new Promise((resolve) => {
+                    const img = new Image();
+                    img.src = path;
+                    img.onload = () => resolve(true);
+                    img.onerror = () => resolve(false);
+                });
+            });
+        }
+        const isValidFileFormat = isValidImage().then(isValidImageResult => {
+            return isValidUrl && isValidImageResult;
+        });
```

Comment:
This matches the technical spec I suggested in the task, thanks!

A few remarks:
- Since we have `isValidUrl`, I think we should avoid fetching if the url is invalid.
- Could you please write a more concrete comment explaining the issue with CORS ?
- I don't like using the second parameter of `then` for the error callback as I think it is harder to read. Use `.then(...).catch(...)`.
- Are you sure your function `isValidImage` returns something? I think it returns nothing. For it to return something it should either have a `return` before `fetch` or having the arrow function return the `fetch` promise directly.
- You are making `validateUrl` return an asynchronous response but the calls to `validateUrl` do not `await` for the result. The function currently returns `{ Promise, boolean}` so I suspect `isValidFileFormat` is considered to be true when the `validateUrl` is called because `Boolean(promise) === true`, even though the actual value to which the promise would resolve is `false`.

Due to `isValidImage` not returning anything and `isValidImage` returning a promise, I don't understand how the current code could work, to be honest.

Reply:
@dmo-odoo  The isValidImage function does indeed return a promise in both cases - when the fetch succeeds and when it fails.I believe i overlooked a scenario when async function isValidImage itself doesn't return anything. The return statements are inside the then and error callback, but not in the isValidImage function itself. This means that when we call `isValidImage().then(...)`, isValidImage() is undefined, and you can't call then on undefined.

The second parameter handles the scenario when promise is rejected, but yeah `.catch` makes it more readable i agree. i will make all the necessary changes. Thanks for those remarks.